rules/Php80/Rector/FunctionLike/UnionTypesRector.php

```diff
@@ -129,6 +137,11 @@ private function refactorParamTypes(FunctionLike $functionLike, PhpDocInfo $phpD
                 continue;
             }
 
+            if ($this->hasObjectWithoutClassType($paramType)) {
+                $this->changeObjectWithoutClassType($paramType, $param, $phpDocInfo, $key);
+                continue;
+            }
```

Comment:
I probably was not clear about this. Could you extract the "narrow `object` and type" to a separated Rector rule?

Reply:
Do you mean this rule skip change when parameter type has object without class type?

Reply:
@TomasVotruba your request on comment https://github.com/rectorphp/rector/pull/6237#issuecomment-826273171 actually implemented in fixture `object_and_specific_class.php.inc` (https://github.com/rectorphp/rector/blob/close-6194/rules-tests/Php80/Rector/FunctionLike/UnionTypesRector/Fixture/object_and_specific_class.php.inc) ,

am I missing something?

Reply:
That's the place I was not clear enough.
This should be a new rule as UnionType should only move the type to PHP, not narrow it down.

Changing docblock by narrowing type should be a new rule. That way we can extend it with other similar rules, e.g.

```diff
-@var int|string|bool|floatarray
+@var scalar
```

Such rule can be applied without PHP 8 as well. It could be named `NarrowUnionTypeDocRector`

Reply:
@TomasVotruba ok, implemented at https://github.com/rectorphp/rector/pull/6237/commits/3e4d3c3d09a4301274afe33b0e36bfe9d2c5876d , skip change `@param` the docblock, only change the type hint param in method.

Reply:
Thanks :+1: 

Now we'll need a new separated rule that narrows the types.

Reply:
I will try.